sessionctx/stmtctx/stmtctx.go

```diff
@@ -60,6 +63,43 @@ type SQLWarn struct {
 	Err   error
 }
 
+type jsonSQLWarn struct {
+	Level  string        `json:"level"`
+	SQLErr *terror.Error `json:"err,omitempty"`
+	Msg    string        `json:"msg,omitempty"`
+}
+
+// MarshalJSON implements the Marshaler.MarshalJSON interface.
+func (warn *SQLWarn) MarshalJSON() ([]byte, error) {
+	w := &jsonSQLWarn{
+		Level: warn.Level,
+	}
+	e := errors.Cause(warn.Err)
+	switch x := e.(type) {
+	case *terror.Error:
+		// Omit outside errors because only the most inside error matters.
```

Comment:
```suggestion
		// Omit outter errors because only the most inner error matters.
```
I think using `inner`/`outter` would be better than `inside`/`outside`?